### ISSUE TITLE
test_read_multiple_no_metadata fails on ubuntu trusty and xenial

### ISSUE THREAD

**Issue by kayibal** [dask/fastparquet#136]
After pip installing fastparquet on a docker container (trusty and xenial) this test seems to fail because the files paths are unsorted:

```
________________________________________________________________________________________________________________________ test_read_multiple_no_metadata ________________________________________________________________________________________________________________________

tempdir = '/tmp/tmpftwrpl9e'

    def test_read_multiple_no_metadata(tempdir):
        df = pd.DataFrame({'x': [1, 5, 2, 5]})
        write(tempdir, df, file_scheme='hive', row_group_offsets=[0, 2])
        os.unlink(os.path.join(tempdir, '_metadata'))
        import glob
        flist = glob.glob(os.path.join(tempdir, '*'))
        pf = ParquetFile(flist)
        assert len(pf.row_groups) == 2
        out = pf.to_pandas()
>       pd.util.testing.assert_frame_equal(out, df)

fastparquet/test/test_api.py:156: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/lib/python3.5/site-packages/pandas/util/testing.py:1313: in assert_frame_equal
    obj='DataFrame.iloc[:, {0}]'.format(i))
/usr/local/lib/python3.5/site-packages/pandas/util/testing.py:1181: in assert_series_equal
    obj='{0}'.format(obj))
pandas/src/testing.pyx:59: in pandas._testing.assert_almost_equal (pandas/src/testing.c:4156)
    ???
pandas/src/testing.pyx:173: in pandas._testing.assert_almost_equal (pandas/src/testing.c:3274)
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = 'DataFrame.iloc[:, 0]', message = 'DataFrame.iloc[:, 0] values are different (50.0 %)', left = '[2, 5, 1, 5]', right = '[1, 5, 2, 5]', diff = None

    def raise_assert_detail(obj, message, left, right, diff=None):
        if isinstance(left, np.ndarray):
            left = pprint_thing(left)
        if isinstance(right, np.ndarray):
            right = pprint_thing(right)
    
        msg = """{0} are different
    
    {1}
    [left]:  {2}
    [right]: {3}""".format(obj, message, left, right)
    
        if diff is not None:
            msg = msg + "\n[diff]: {diff}".format(diff=diff)
    
>       raise AssertionError(msg)
E       AssertionError: DataFrame.iloc[:, 0] are different
E       
E       DataFrame.iloc[:, 0] values are different (50.0 %)
E       [left]:  [2, 5, 1, 5]
E       [right]: [1, 5, 2, 5]

/usr/local/lib/python3.5/site-packages/pandas/util/testing.py:1018: AssertionError
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
====================================================================================================================== 1 failed, 8 passed in 1.55 seconds ======================================================================================================================
```

**Comment by martindurant** [comment]
Does the failure go away if 
```
flist = glob.glob(os.path.join(tempdir, '*'))
->
flist = list(sorted(glob.glob(os.path.join(tempdir, '*'))))
```
?

**Comment by kayibal** [comment]
yes

**Comment by martindurant** [comment]
Fixes in #135
